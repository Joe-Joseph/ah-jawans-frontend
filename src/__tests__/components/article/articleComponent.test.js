import React from 'react';
import { shallow, mount } from '../../../../config/enzymeConfig';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import CreateArticle from '../../../components/article/ArticleComponent';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  article: { article: {} },
  userCredentials: { userCredentials: {} },
});

const props = {
  article: jest.fn(),
};

// mock for mutations observer window object
global.MutationObserver = class {
  disconnect() {}
  observe() {}
};

describe('<CreateArticle />', () => {
  beforeEach(() => { });

  afterEach(() => {
    jest.clearAllMocks();
  });
  const component = shallow(
    <Provider store={store}>
      <CreateArticle {...props} />
    </Provider>,
  );

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('Should render the component', () => {
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <CreateArticle />
        </Router>
      </Provider>,
    );
    expect(wrapper.find('Editor')).toBeDefined();
  });
});
