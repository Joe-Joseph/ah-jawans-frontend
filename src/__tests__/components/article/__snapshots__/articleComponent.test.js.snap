// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateArticle /> should render without crashing 1`] = `ShallowWrapper {}`;
